x/mongo/driver/operation/delete.go

```diff
@@ -54,10 +55,13 @@ func buildDeleteResult(response bsoncore.Document) (DeleteResult, error) {
 	for _, element := range elements {
 		switch element.Key() {
 		case "n":
-			var ok bool
-			dr.N, ok = element.Value().AsInt32OK()
```

Comment:
Consider using `AsInt64OK()` to simplify here and in insert.go and update.go. `AsInt64OK()` also accepts Double. But I do not see any harm in accepting Double.

Reply:
Whoa, I didn't realize these are actually using a mixture of `AsInt__OK` and `Int__OK` and that `AsInt__OK` converts types already 🤯 . Great suggestion, will update all to use `AsInt64OK` everywhere!